docker/Dockerfile

```diff
@@ -1,10 +1,27 @@
-FROM clux/muslrust:1.67.0 as builder
+FROM clux/muslrust:1.67.0 AS chef
+USER root
+RUN cargo install cargo-chef
 WORKDIR /app
+
+FROM chef AS planner
+COPY . .
+RUN cargo chef prepare --recipe-path recipe.json
+
+FROM chef AS builder
+
 ARG CARGO_BUILD_TARGET=x86_64-unknown-linux-musl
 
 # This can be set to release using --build-arg
 ARG RUST_RELEASE_MODE="debug"
 
+COPY --from=planner /app/recipe.json recipe.json
+RUN if [ "$RUST_RELEASE_MODE" = "debug" ] ; then \
```

Comment:
Did these really have any effect? On line 30 it uses the native docker cache.

Reply:
From my testing, yes, it did have an effect.  
I didn't remove the other lines just in case it was also helping in some way.  

From some examples I've seen the line 30 is kind of incomplete, it should look something like this  
```
RUN --mount=type=cache,target=/usr/local/cargo,from=rust:1.68.1,source=/usr/local/cargo \
    --mount=type=cache,target=target \
    cargo build
```
I tried it but not very extensively because I wasn't seeing improvements and I focused on cargo-chef.  
In theory the missing parts from the current implementation would be to check where to point `source` in the muslrust container.

Reply:
Can you post some benchmarks from your tests? In particular compare the following cases between the current setup and using docker-chef: 
- clean build
- incremental build after changing a file in `src/`
- incremental build after adding a new dependency

Reply:
I'll test this locally now.

Reply:
This should also be fixed by updating cargo in docker to 1.70 so give that a try.